azext_iot/common/shared.py

```diff
@@ -267,3 +267,20 @@ class IoTHubStateType(Enum):
     KeyEncryptionKeyRevoking = "KeyEncryptionKeyRevoking"
     KeyEncryptionKeyRevoked = "KeyEncryptionKeyRevoked"
     ReActivating = "ReActivating"
+
+
+class ConnectionStringParser(Enum):
+    """
+        All connection string parser with respective functions
+    """
+    from azext_iot.common._azure import (
+        parse_iot_device_connection_string,
+        parse_iot_device_module_connection_string,
+        parse_pnp_connection_string,
+        parse_iot_hub_connection_string
+    )
+
+    Module = parse_iot_device_module_connection_string
+    Device = parse_iot_device_connection_string
+    PnP = parse_pnp_connection_string
```

Comment:
`parse_pnp_connection_string` can be completely eliminated from the codebase. It's a legacy remnant from a prior version of the PnP model repository.